File: backend/BowlerAPI/Data/BowlerScore.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BowlerAPI.Data;

public partial class BowlerScore
{
    [Key]
    [Required]
    public int MatchId { get; set; }
    [Key]
    [Required]
    public short GameNumber { get; set; }
    [Key]
    [ForeignKey("Bowler")]
    [Required]
    public int BowlerId { get; set; }

    public short? RawScore { get; set; }

    public short? HandiCapScore { get; set; }

    public bool WonGame { get; set; }
    [Required]
    public virtual Bowler Bowler { get; set; } = null!;
}

File: frontend1/src/types/BowlerScore.ts
import { Bowler } from './Bowler';

export type BowlerScores = {
  MatchId: number;
  GameNumber: number;
  BowlerId: number;
  RawScore: number;
  HandiCapScore: number;
  WonGame: boolean;
  Bowler: Bowler;
};
